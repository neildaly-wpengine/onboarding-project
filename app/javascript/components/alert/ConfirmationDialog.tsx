import {
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  DialogTitle,
  Typography,
} from "@material-ui/core";
import React from "react";

interface ConfirmationDialogProps {
  open: boolean;
  handleCancel(): void;
  handleOk(): void;
  title: string;
  message: string;
}

const ConfirmationDialog: React.FC<ConfirmationDialogProps> = ({
  open,
  handleCancel,
  handleOk,
  title,
  message,
}) => {
  return (
    <Dialog
      data-testid="delete-dialog"
      disableBackdropClick
      disableEscapeKeyDown
      maxWidth="xs"
      aria-labelledby="confirmation-dialog-title"
      open={open}
    >
      <DialogTitle id="confirmation-dialog-title">{title}</DialogTitle>
      <DialogContent>
        <Typography variant="body1" data-testid="dialog-message">
          {message}
        </Typography>
      </DialogContent>
      <DialogActions>
        <Button
          autoFocus
          onClick={handleCancel}
          color="primary"
          data-testid="dialog-cancel"
        >
          Cancel
        </Button>
        <Button onClick={handleOk} color="primary" data-testid="dialog-ok">
          Ok
        </Button>
      </DialogActions>
    </Dialog>
  );
};

export default ConfirmationDialog;
