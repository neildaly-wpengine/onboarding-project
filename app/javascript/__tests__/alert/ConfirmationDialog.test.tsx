import "@testing-library/jest-dom/extend-expect";
import {
  cleanup,
  render,
  RenderResult,
  act,
  fireEvent,
} from "@testing-library/react";
import * as React from "react";
import ConfirmationDialog from "../../components/alert/ConfirmationDialog";

const renderHelper = (
  open = false,
  handleCancel = jest.fn(),
  handleOk = jest.fn()
): RenderResult => {
  return render(
    <ConfirmationDialog
      open={open}
      handleCancel={handleCancel}
      handleOk={handleOk}
      title="Some Title"
      message="Some message.."
    />
  );
};

afterEach(cleanup);

describe("<ConfirmationDialog />", () => {
  describe("Snapshots", () => {
    test("should not render a confirmation dialog when closed", () => {
      const { container } = renderHelper();

      expect(container).toMatchSnapshot();
    });

    test("should render a confirmation dialog when open", () => {
      const { getByTestId } = renderHelper(true);

      expect(getByTestId("delete-dialog")).toBeInTheDocument();
    });
  });
  describe("Actions", () => {
    test("should fire appropriate actions when clicked", () => {
      const closeAction: jest.Mock = jest.fn(),
        okAction: jest.Mock = jest.fn();
      const { getByTestId } = renderHelper(true, closeAction, okAction);

      act(() => {
        fireEvent.click(getByTestId("dialog-cancel"));
        fireEvent.click(getByTestId("dialog-ok"));
      });

      expect(closeAction).toHaveBeenCalledTimes(1);
      expect(okAction).toHaveBeenCalledTimes(1);
    });
  });
});
