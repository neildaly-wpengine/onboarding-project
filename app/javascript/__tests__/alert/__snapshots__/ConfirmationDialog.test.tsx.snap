// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConfirmationDialog /> Snapshots should not render a confirmation dialog when closed 1`] = `<div />`;
